account.h

```diff
@@ -0,0 +1,108 @@
+#ifndef account_H
+#define account_H
+
+using json = nlohmann::json;
+
+class account {
+protected:
+    int balance{};
+    std::string id;
+    std::string name;
+    std::string surname;
+
+public:
+    int getBalance() {
+        return balance;
+    }
+    std::string getID() {
+        return id;
+    }
+    std::string getName() {
+        return name;
+    }
+    std::string getSurname() {
+        return surname;
+    }
+
+    void CreateAccount(account& a);
+    void UpdateAccount(account& a, const int& id_pos);
+    account(std::string i, std::string n, std::string s, int b);
+};
+
+account::account(std::string i, std::string n, std::string s, int b)
+{
+    id = i;
+    balance = b;
+    name = n;
+    surname = s;
+}
+
+void account::CreateAccount(account& a)
+{
+    json j_new;
+
+    std::ifstream old_file("CustomerList.json");
+    if (old_file.is_open())
+    {
+        try {
+            old_file >> j_new;
+        }
+        catch (...) {
+            std::cout << "Customer list initialized.\n";
+            j_new["CustomerList"] = json::array({});
```

Comment:
if there is exception in 48 line, than you do not have j_new["CustomerList"]

Reply:
This was a treatment for when a file does not exist. I couldn't find a way to check if the file contains error or not, because the ifstream only throws one type of exception for both scenarios, so in this case if the file does not exist and there is no j_new["CustomerList"], one would be initialized by lines 51-52. Is there something else that I am missing from your comment?

Reply:
OK